src/openzeppelin/token/erc721/dual721.cairo

```diff
@@ -0,0 +1,195 @@
+use traits::Into;
+use traits::TryInto;
+use array::SpanTrait;
+use array::ArrayTrait;
+use option::OptionTrait;
+use starknet::ContractAddress;
+use starknet::SyscallResultTrait;
+use starknet::call_contract_syscall;
+use starknet::Felt252TryIntoContractAddress;
+use openzeppelin::token::erc721::interface::IERC721;
+use openzeppelin::utils::try_selector_with_fallback;
+use openzeppelin::utils::Felt252TryIntoBool;
+use openzeppelin::utils::BoolIntoFelt252;
+use openzeppelin::utils::constants;
+
+#[derive(Copy, Drop)]
+struct DualERC721 {
```

Comment:
```suggestion
struct DualCaseERC721Caller {
```

I think that, from the name, is hard to guess that this is meant to be a casing agnostic caller. I think the suggestion could improve readability.

Reply:
thanks, improving the names is an item in my to do list, for this module and for the mock contracts as well.

I'm not sure about `Caller`, i want this module to be transparent to users: to look and feel like interacting directly with an ERC271 (or with `call_contract_syscall`). therefore not calling it a caller to me helps convey that intuition.

Reply:
`DualCaseERC721` could work. in a way it's also something that "casts" or converts something to be interacted with snake, hmm. We can't call it `SnakeERC721` since it could confuse people into thinking it's only for snake_case. `CaseNormalizedERC721` is too long. 

i guess `DualCase` might be the best.

Reply:
I added "Caller" because the inner address is represented as a target, so the concept of a caller "calling" the target made sense in my mind. If as you said, you want this to be like an abstraction, I would call the address field something like: inner, or contract_addr... not sure actually. Naming things :(